staging/src/k8s.io/apiextensions-apiserver/pkg/registry/customresource/etcd_test.go

```diff
@@ -160,6 +163,64 @@ func TestDelete(t *testing.T) {
 	test.TestDelete(validNewCustomResource())
 }
 
+func TestGeneration(t *testing.T) {
+	// enable alpha feature CustomResourceSubresources
+	defer utilfeaturetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.CustomResourceSubresources, true)()
+
+	storage, server := newStorage(t)
+	defer server.Terminate(t)
+	defer storage.CustomResource.Store.DestroyFunc()
+	modifiedRno := *validNewCustomResource()
+	modifiedRno.SetGeneration(10)
+	setObservedGeneration(&modifiedRno, 10)
+	ctx := genericapirequest.NewDefaultContext()
+	cr, err := createCustomResource(storage.CustomResource, modifiedRno, t)
+	if err != nil {
+		t.Errorf("unexpected error: %v", err)
+	}
+	etcdCR, err := storage.CustomResource.Get(ctx, cr.GetName(), &metav1.GetOptions{})
+	if err != nil {
+		t.Errorf("unexpected error: %v", err)
+	}
+	storedCR, _ := etcdCR.(*unstructured.Unstructured)
+
+	// Generation initialization
+	if storedCR.GetGeneration() != 1 || getObservedGeneration(storedCR) != 0 {
+		t.Fatalf("Unexpected generation number %v, status generation %v", storedCR.GetGeneration(), getObservedGeneration(storedCR))
+	}
+
+	// Updates to spec should increment the generation number
+	setSpecReplicas(storedCR, getSpecReplicas(storedCR)+1)
+	storage.CustomResource.Update(ctx, storedCR.GetName(), rest.DefaultUpdatedObjectInfo(storedCR), rest.ValidateAllObjectFunc, rest.ValidateAllObjectUpdateFunc)
```

Comment:
need to equate to `err` here

Reply:
same update needs to be done for the replicaset + replicationcontroller test.

Reply:
Done. There is a equate to err in replicationcontroller test yet, and I add equate to err in replicaset test.